source/application/DbgHelp/DbgHelpStackWalkCallback.cpp

```diff
@@ -207,13 +351,8 @@ void DbgHelpStackWalkCallback::EnumerateParameters()
          * This will be incremented for each found symbol, with it's size.
          */
         DWORD offset = 0;
-        for (DWORD count = 0; (enum_symbols->Next(1, &sym_info, &num_fetched) == S_OK); count++)
+        for (DWORD count = 0; (enum_symbols->Next(1, &sym_info, &num_fetched) == S_OK && num_fetched == 1); count++)

```

Comment:
Could also write a nice C++ iterator. I'll try something like that and see if it feels good.